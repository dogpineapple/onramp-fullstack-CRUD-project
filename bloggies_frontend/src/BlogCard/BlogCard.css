.BlogCard {
  height: fit-content;
  max-height: 20rem;
  margin: 0.5rem 0;
}

.BlogCard-body {
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

.BlogCard-author-date {
  padding-top: 1rem;
}

