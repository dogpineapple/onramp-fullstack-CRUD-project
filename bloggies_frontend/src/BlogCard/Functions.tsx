import React, { useState } from 'react'
import axios from 'axios'

//Functionality for payment success and for payment cancelation 


interface FormData {
    answer1: string    
    answer2: string 
    answer3: string 

}




const formPost = (formData: FormData) => {
     // goes on register form to post membership registration status
     return axios.post('/register/membership-status')
     //if successful then direct user to success page if not successful then show pending message
     //after form message is posted to db server checks to ensure answers are correct then 
     //another func sends SendGrid email to users email  
 }




const onSubmit = async () => {
    try {
        await formPost({answer1: '', answer2: '', answer3: ''})
    } catch(error) {
        console.log(error)
    }   
}




