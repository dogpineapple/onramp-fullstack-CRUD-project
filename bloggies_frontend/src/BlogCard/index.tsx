import moment from "moment";
import React from "react";
import { Card, Container, Row } from "react-bootstrap";
import { NavLink } from "react-router-dom";
import { Post } from "../custom";
import { changeToURLFriendly } from "../helpers";
import FavoriteButton from "../FavoriteButton";
import "./BlogCard.css";

interface IProp {
  post: Post
}

function BlogCard({ post }: IProp) {
  const postTitleForURL = changeToURLFriendly(post.title);

  return (
    <Card className="BlogCard text-left">
      <Card.Body>
        <NavLink to={`/blogs/${post.id}/${postTitleForURL}`}>
          <Card.Title>{post.title}</Card.Title>
          <Card.Subtitle>{post.description}</Card.Subtitle>
          <Card.Text className="BlogCard-body">
            {post.body}
          </Card.Text>
        </NavLink>
        <Container fluid>
          <Row className="d-flex align-items-baseline justify-content-between">
            <Card.Subtitle className="BlogCard-author-date">Posted by {post.author_name} <span className="text-muted"> {moment(post.created_at).fromNow()}</span> {post.last_updated_at !== post.created_at && <span className="App-update"> (last updated {moment(post.last_updated_at).fromNow()})</span>}</Card.Subtitle>
            <FavoriteButton post={post}></FavoriteButton>
          </Row>
        </Container>
      </Card.Body>
    </Card>
  );
};

export default BlogCard;