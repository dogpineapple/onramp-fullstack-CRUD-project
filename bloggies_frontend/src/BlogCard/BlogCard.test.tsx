import React, { Component } from 'react';
import Enzyme, { mount, ReactWrapper, shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import BlogCard from './index';
import ReactRouter from 'react-router';
import { MOCK_POST, MOCK_STORE } from '../jest.mock';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

Enzyme.configure({ adapter: new Adapter() });

const mockStore = configureStore();

describe('BlogCard', () => {
  let wrapper: ReactWrapper;

  beforeEach(() => {
    jest.spyOn(ReactRouter, 'useParams').mockReturnValue({ userId: '1', displayName: 'test-user' });

    wrapper = mount(shallow<Component>(
      <Provider store={mockStore(MOCK_STORE)}>
        <BrowserRouter>
          <BlogCard post={MOCK_POST} />
        </BrowserRouter>
      </Provider>
    ).get(0));
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should display correct post text', () => {
    const bodyTextTag = wrapper.find(".BlogCard-body > p");
    expect(bodyTextTag.text()).toBe(MOCK_POST.body);
  });
});