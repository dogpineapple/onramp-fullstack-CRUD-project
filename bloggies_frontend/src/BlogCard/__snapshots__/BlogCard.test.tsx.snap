// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogCard should match snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <BlogCard
      post={
        Object {
          "author_id": 1,
          "author_name": "test user",
          "body": "test body",
          "bookmark_count": "5",
          "created_at": "1/1/2020",
          "description": "test description",
          "id": 123,
          "is_premium": false,
          "last_updated_at": "1/1/2020",
          "title": "test title",
        }
      }
    >
      <Card
        body={false}
        className="BlogCard text-left"
      >
        <div
          className="BlogCard text-left card"
        >
          <CardBody>
            <div
              className="card-body"
            >
              <Styled(Component)
                isPremium={false}
                to="/blogs/123/test-title"
              >
                <Component
                  className="sc-bdfBwQ lnRVtZ"
                  isPremium={false}
                  to="/blogs/123/test-title"
                >
                  <NavLink
                    className="sc-bdfBwQ lnRVtZ"
                    to="/blogs/123/test-title"
                  >
                    <Link
                      aria-current={null}
                      className="sc-bdfBwQ lnRVtZ"
                      to={
                        Object {
                          "hash": "",
                          "pathname": "/blogs/123/test-title",
                          "search": "",
                          "state": null,
                        }
                      }
                    >
                      <LinkAnchor
                        aria-current={null}
                        className="sc-bdfBwQ lnRVtZ"
                        href="/blogs/123/test-title"
                        navigate={[Function]}
                      >
                        <a
                          aria-current={null}
                          className="sc-bdfBwQ lnRVtZ"
                          href="/blogs/123/test-title"
                          onClick={[Function]}
                        >
                          <CardTitle>
                            <ForwardRef
                              className="card-title"
                            >
                              <div
                                className="card-title h5"
                              >
                                test title
                              </div>
                            </ForwardRef>
                          </CardTitle>
                          <CardSubtitle>
                            <ForwardRef
                              className="card-subtitle"
                            >
                              <div
                                className="card-subtitle h6"
                              >
                                test description
                              </div>
                            </ForwardRef>
                          </CardSubtitle>
                          <CardText
                            className="BlogCard-body"
                          >
                            <p
                              className="BlogCard-body card-text"
                            >
                              test body
                            </p>
                          </CardText>
                        </a>
                      </LinkAnchor>
                    </Link>
                  </NavLink>
                </Component>
              </Styled(Component)>
              <Container
                fluid={true}
              >
                <div
                  className="container-fluid"
                >
                  <Row
                    className="d-flex align-items-baseline justify-content-between"
                    noGutters={false}
                  >
                    <div
                      className="d-flex align-items-baseline justify-content-between row"
                    >
                      <CardSubtitle
                        className="BlogCard-author-date d-flex align-items-center"
                      >
                        <ForwardRef
                          className="BlogCard-author-date d-flex align-items-center card-subtitle"
                        >
                          <div
                            className="BlogCard-author-date d-flex align-items-center card-subtitle h6"
                          >
                            Posted by 
                            <Styled(NavLink)
                              className="d-flex align-items-center ml-1 mr-1"
                              to="/users/1/test-user"
                            >
                              <NavLink
                                className="sc-gsTCUz kyRshY d-flex align-items-center ml-1 mr-1"
                                to="/users/1/test-user"
                              >
                                <Link
                                  aria-current={null}
                                  className="sc-gsTCUz kyRshY d-flex align-items-center ml-1 mr-1"
                                  to={
                                    Object {
                                      "hash": "",
                                      "pathname": "/users/1/test-user",
                                      "search": "",
                                      "state": null,
                                    }
                                  }
                                >
                                  <LinkAnchor
                                    aria-current={null}
                                    className="sc-gsTCUz kyRshY d-flex align-items-center ml-1 mr-1"
                                    href="/users/1/test-user"
                                    navigate={[Function]}
                                  >
                                    <a
                                      aria-current={null}
                                      className="sc-gsTCUz kyRshY d-flex align-items-center ml-1 mr-1"
                                      href="/users/1/test-user"
                                      onClick={[Function]}
                                    >
                                      test user
                                    </a>
                                  </LinkAnchor>
                                </Link>
                              </NavLink>
                            </Styled(NavLink)>
                            <span
                              className="text-muted"
                            >
                               
                              a year ago
                            </span>
                             
                          </div>
                        </ForwardRef>
                      </CardSubtitle>
                      <FavoriteButton
                        post={
                          Object {
                            "author_id": 1,
                            "author_name": "test user",
                            "body": "test body",
                            "bookmark_count": "5",
                            "created_at": "1/1/2020",
                            "description": "test description",
                            "id": 123,
                            "is_premium": false,
                            "last_updated_at": "1/1/2020",
                            "title": "test title",
                          }
                        }
                      >
                        <div
                          className="FavoriteButton d-flex align-items-center"
                        >
                          <FontAwesomeIcon
                            border={false}
                            className="FavoriteButton-btn"
                            fixedWidth={false}
                            flip={null}
                            icon={
                              Object {
                                "icon": Array [
                                  512,
                                  512,
                                  Array [],
                                  "f004",
                                  "M458.4 64.3C400.6 15.7 311.3 23 256 79.3 200.7 23 111.4 15.6 53.6 64.3-21.6 127.6-10.6 230.8 43 285.5l175.4 178.7c10 10.2 23.4 15.9 37.6 15.9 14.3 0 27.6-5.6 37.6-15.8L469 285.6c53.5-54.7 64.7-157.9-10.6-221.3zm-23.6 187.5L259.4 430.5c-2.4 2.4-4.4 2.4-6.8 0L77.2 251.8c-36.5-37.2-43.9-107.6 7.3-150.7 38.9-32.7 98.9-27.8 136.5 10.5l35 35.7 35-35.7c37.8-38.5 97.8-43.2 136.5-10.6 51.1 43.1 43.5 113.9 7.3 150.8z",
                                ],
                                "iconName": "heart",
                                "prefix": "far",
                              }
                            }
                            inverse={false}
                            listItem={false}
                            mask={null}
                            onClick={[Function]}
                            pull={null}
                            pulse={false}
                            rotation={null}
                            size="1x"
                            spin={false}
                            swapOpacity={false}
                            symbol={false}
                            title=""
                            transform={null}
                          >
                            <svg
                              aria-hidden="true"
                              className="svg-inline--fa fa-heart fa-w-16 fa-1x FavoriteButton-btn"
                              data-icon="heart"
                              data-prefix="far"
                              focusable="false"
                              onClick={[Function]}
                              role="img"
                              style={Object {}}
                              viewBox="0 0 512 512"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M458.4 64.3C400.6 15.7 311.3 23 256 79.3 200.7 23 111.4 15.6 53.6 64.3-21.6 127.6-10.6 230.8 43 285.5l175.4 178.7c10 10.2 23.4 15.9 37.6 15.9 14.3 0 27.6-5.6 37.6-15.8L469 285.6c53.5-54.7 64.7-157.9-10.6-221.3zm-23.6 187.5L259.4 430.5c-2.4 2.4-4.4 2.4-6.8 0L77.2 251.8c-36.5-37.2-43.9-107.6 7.3-150.7 38.9-32.7 98.9-27.8 136.5 10.5l35 35.7 35-35.7c37.8-38.5 97.8-43.2 136.5-10.6 51.1 43.1 43.5 113.9 7.3 150.8z"
                                fill="currentColor"
                                style={Object {}}
                              />
                            </svg>
                          </FontAwesomeIcon>
                          <span>
                            5
                          </span>
                        </div>
                      </FavoriteButton>
                    </div>
                  </Row>
                </div>
              </Container>
            </div>
          </CardBody>
        </div>
      </Card>
    </BlogCard>
  </Router>
</BrowserRouter>
`;
